pkg/k8shandler/visualization.go

```diff
@@ -432,32 +436,75 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaConsoleExternal
 		return err
 	}
 
-	consoleExternalLogLink := NewConsoleExternalLogLink(
-		"kibana",
+	consoleExternalLogLinkForAdmin := NewConsoleExternalLogLink(
+		"kibana-admin",
+		cluster.Namespace,
+		"Show in Kibana",
+		strings.Join([]string{kibanaURL,
+			"/app/kibana#/discover?_g=(time:(from:now-1w,mode:relative,to:now))&_a=(columns:!(kubernetes.container_name,message),index:",
+			"'.operations.*'",
+			",query:(query_string:(analyze_wildcard:!t,query:'",
+			strings.Join([]string{
+				"kubernetes.pod_name:\"${resourceName}\"",
+				"kubernetes.namespace_name:\"${resourceNamespace}\"",
+				"kubernetes.container_name.raw:\"${containerName}\"",
+			}, " AND "),
+			"')),sort:!('@timestamp',desc))"},
+			""),
+		adminNamespaceFilter,
+	)
+
+	consoleExternalLogLinkForNonAdmin := NewConsoleExternalLogLink(
+		"kibana-non-admin",
 		cluster.Namespace,
 		"Show in Kibana",
 		strings.Join([]string{kibanaURL,
-			"/app/kibana#/discover?_g=(time:(from:now-1w,mode:relative,to:now))&_a=(columns:!(kubernetes.container_name,message),query:(query_string:(analyze_wildcard:!t,query:'",
+			"/app/kibana#/discover?_g=(time:(from:now-1w,mode:relative,to:now))&_a=(columns:!(kubernetes.container_name,message),index:",
+			"'project.${kubernetes.namespace_name}.${resourceNamespaceUID}.*'",
+			",query:(query_string:(analyze_wildcard:!t,query:'",
 			strings.Join([]string{
 				"kubernetes.pod_name:\"${resourceName}\"",
 				"kubernetes.namespace_name:\"${resourceNamespace}\"",
 				"kubernetes.container_name.raw:\"${containerName}\"",
 			}, " AND "),
 			"')),sort:!('@timestamp',desc))"},
 			""),
+		nonAdminNamespaceFilter,
 	)
 
-	utils.AddOwnerRefToObject(consoleExternalLogLink, utils.AsOwner(cluster))
+	links := []*consolev1.ConsoleExternalLogLink{ consoleExternalLogLinkForAdmin, consoleExternalLogLinkForNonAdmin}
 
-	// In case the object already exists we delete it first
-	if err = clusterRequest.RemoveConsoleExternalLogLink("kibana"); err != nil {
-		return
+	for _, externalLink := range links {
+		utils.AddOwnerRefToObject(externalLink, utils.AsOwner(cluster))
+		// In case the object already exists we delete it first
+		if err = clusterRequest.RemoveConsoleExternalLogLink(externalLink.ObjectMeta.Name); err != nil {
+			return
+		}
+		err = clusterRequest.Create(externalLink)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("Failure creating Kibana console external %q log link for %q: %v",
+				externalLink.ObjectMeta.Name, cluster.Name, err)
+		}
 	}
+	//utils.AddOwnerRefToObject(consoleExternalLogLinkForAdmin, utils.AsOwner(cluster))
+	//utils.AddOwnerRefToObject(consoleExternalLogLinkForNonAdmin, utils.AsOwner(cluster))
 
-	err = clusterRequest.Create(consoleExternalLogLink)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana console external log link for %q: %v", cluster.Name, err)
-	}
+	// In case the object already exists we delete it first
```

Comment:
can we remove this if it is not required

Reply:
Yes, we can. And I will remove it. The only reason I introduced this logic is to allow CLO re-create the external link objects on restart (aka `make run`) so that all code changes for the external links are reflected and can be immediately tested.
The only other alternative would be to delete relevant objects in K8s manually before doing `make run`, or is there other way?

Reply:
it is still required for normal operator reconciling... when we initially created this logic it was noticed that we could not do an `update` because some fields that we would possibly change were immutable so we need to do a `delete` and then a `create` every time.